src/github3/checks.py

```diff
@@ -0,0 +1,539 @@
+
+    .. attribute:: app
+
+        A :class:`~github3.checks.CheckApp` representing the App
+        this run belongs to.
+
+    .. CheckRun Documentation:
+        http://developer.github.com/v3/checks/runs/
+    """
+
+    class_name = "CheckRun"
+    list_response_dict_key = "check_runs"
+    CUSTOM_HEADERS = {"Accept": "application/vnd.github.antiope-preview+json"}
+
+    def _update_attributes(self, run):
+        self._api = run["url"]
+        self.html_url = run["html_url"]
+        self.status = run["status"]
+        self.conclusion = run["conclusion"]
+        self.started_at = self._strptime(run["started_at"])
+        self.completed_at = self._strptime(run["completed_at"])
+        self.head_sha = run["head_sha"]
+        self.name = run["name"]
+        prs = run.get("pull_requests", [])
+        self.original_pull_requests = [CheckPullRequest(p, self) for p in prs]
+        self.id = run["id"]
+        self.external_id = run["external_id"]
+        self.app = CheckApp(run["app"], self)
+        self.check_suite = run["check_suite"]["id"]
+        self.output = CheckRunOutput(run["output"], self)
+
+    def _repr(self):
+        return "<{s.class_name} [{s.name}:{s.status}]>".format(s=self)
+
+    @decorators.requires_app_installation_auth
+    def update(
+        self,
+        name=None,
+        details_url=None,
+        external_id=None,
+        started_at=None,
+        status=None,
+        conclusion=None,
+        completed_at=None,
+        output=None,
+        actions=None,
+    ):
+        """Update this check run.
+
+        All parameters are optional.
+
+        :param str name:
+            (optional), new name of the check
+        :param str details_url:
+            (optional), The URL of the integrator's site that has the full
+            details of the check
+        :param str external_id:
+            (optional), A reference for the run on the integrator's system
+        :param str started_at:
+            (optional), ISO 8601 time format: YYYY-MM-DDTHH:MM:SSZ
+        :param str status:
+            (optional), ('queued', 'in_progress', 'completed')
+        :param str conclusion:
+            (optional), Required if you provide 'completed_at', or a
+            'status' of 'completed'. The final conclusion of the check.
+            ('success', 'failure', 'neutral', 'cancelled', 'timed_out',
+            'action_required')
+        :param str completed_at:
+            (optional), Required if you provide 'conclusion'. ISO 8601 time
+            format: YYYY-MM-DDTHH:MM:SSZ
+        :param dict output:
+            (optional), key-value pairs representing the output. Format:
+            {'title': 'string', 'summary', 'text, can be markdown', 'text':
+            'text, can be markdown', 'annotations': [{}], 'images': [{}]}
+        :param array actions:
+            (optional), array of action objects. Object format is:
+            {'label': 'text', 'description', 'text', 'identifier', 'text'}
```

Comment:
This seems like it should be code-formatted in the docs, e.g., 

```rst
``{'label': 'text', 'description', ...}``
```

Also this is described as an `array` but looks like a dictionary here. Can you clarify what you mean by "array" or confirm that it is a dictionary and fix the example?

Reply:
It's a list/array of action objects with the action object format as described. I will try to improve this so it's more clear.